src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLSurfaceDataBase.h

```diff
@@ -0,0 +1,163 @@
+ * the underlying surface (e.g. pbuffer) has an alpha channel and isOpaque
+ * is true, then we should take appropriate action (i.e. call glColorMask()
+ * to disable writes into the alpha channel) to ensure that the surface
+ * remains fully opaque.
+ *
+ *     jboolean needsInit;
+ * If true, the surface requires some one-time initialization, which should
+ * be performed after a context has been made current to the surface for
+ * the first time.
+ *
+ *     jint x/yOffset
+ * The offset in pixels of the Metal viewport origin from the lower-left
+ * corner of the heavyweight drawable.  For example, a top-level frame on
+ * Windows XP has lower-left insets of (4,4).  The Metal viewport origin
+ * would typically begin at the lower-left corner of the client region (inside
+ * the frame decorations), but AWT/Swing will take the insets into account
+ * when rendering into that window.  So in order to account for this, we
+ * need to adjust the Metal viewport origin by an x/yOffset of (-4,-4).  On
+ * X11, top-level frames typically don't have this insets issue, so their
+ * x/yOffset would be (0,0) (the same applies to pbuffers).
+ *
+ *     jint width/height;
+ * The cached surface bounds.  For offscreen surface types (MTLSD_FBOBJECT,
+ * MTLSD_TEXTURE, etc.) these values must remain constant.  Onscreen window
+ * surfaces (MTLSD_WINDOW, MTLSD_FLIP_BACKBUFFER, etc.) may have their
+ * bounds changed in response to a programmatic or user-initiated event, so
+ * these values represent the last known dimensions.  To determine the true
+ * current bounds of this surface, query the native Drawable through the
+ * privOps field.
+ *
+ *     void* pTexture;
+ * The texture object handle, as generated by MTLTextureDescriptor(). If this
+ * value is null, the texture has not yet been initialized.
+ *
+ *     jint textureWidth/Height;
+ * The actual bounds of the texture object for this surface.  If the
+ * GL_ARB_texture_non_power_of_two extension is not present, the dimensions
+ * of an OpenGL texture object must be a power-of-two (e.g. 64x32 or 128x512).
+ * The texture image that we care about has dimensions specified by the width
+ * and height fields in this MTLSDOps structure.  For example, if the image
+ * to be stored in the texture has dimensions 115x47, the actual OpenGL
+ * texture we allocate will have dimensions 128x64 to meet the pow2
+ * restriction.  The image bounds within the texture can be accessed using
+ * floating point texture coordinates in the range [0.0,1.0].
+ */
+typedef struct {
+    SurfaceDataOps               sdOps;
+    void                         *privOps;
+    jobject                      graphicsConfig;
+    jint                         drawableType;
+    jint                       activeBuffer;
+    jboolean                     isOpaque;
+    jboolean                     needsInit;
+    jint                         xOffset;
+    jint                         yOffset;
```

Comment:
Are all of these props actually used? Or copied from the "_OGLSDOps"?

Reply:
Verified  activeBuffer is not used. So removing it under JDK-8260640. All others looks like are used in code after searching. Dimensions looks to be redundant.